File: Daily Challanges/2026-07-July/2026-07-09.js
function triageIssue(title, labels) {
  const t = title.toLowerCase();
  labels = [...labels];

  if (labels.length === 0) {
    if (t.includes("error") || t.includes("bug")) {
      labels.push("bug", "needs triage");
    } else if (t.includes("feature") || t.includes("add")) {
      labels.push("enhancement", "discussing");
    }
  } else {
    if (labels.includes("needs triage")) {
      labels = labels.filter((x) => x !== "needs triage");
      if (t.includes("simple") || t.includes("easy")) {
        labels.push("good first issue");
      } else {
        labels.push("help wanted");
      }
    } else if (labels.includes("discussing")) {
      labels = labels.filter((x) => x !== "discussing");
      if (t.includes("planned") || t.includes("next")) {
        labels.push("on the roadmap");
      } else {
        labels.push("help wanted");
      }
    }
  }

  if (t.includes("security") && !labels.includes("critical")) {
    labels.push("critical");
  }

  return labels;
}

console.log(triageIssue("app crashes with error", []));
console.log(triageIssue("app crashes with error", ["bug", "needs triage"]));
console.log(triageIssue("add dark mode", []));
console.log(triageIssue("add dark mode", ["enhancement", "discussing"]));
console.log(triageIssue("xss security bug", []));
console.log(triageIssue("security vulnerability in auth", []));
console.log(triageIssue("easy a11y fix", ["bug", "needs triage"]));
console.log(
  triageIssue("planned api migration", ["enhancement", "discussing"]),
);
console.log(triageIssue("improve security", ["enhancement", "discussing"]));

File: Daily Challanges/2026-07-July/2026-07-09.py
def triage_issue(title, labels):
    t = title.lower()
    labels = labels[:]

    if not labels:
        if "error" in t or "bug" in t:
            labels.extend(["bug", "needs triage"])
        elif "feature" in t or "add" in t:
            labels.extend(["enhancement", "discussing"])
    else:
        if "needs triage" in labels:
            labels.remove("needs triage")
            if "simple" in t or "easy" in t:
                labels.append("good first issue")
            else:
                labels.append("help wanted")
        elif "discussing" in labels:
            labels.remove("discussing")
            if "planned" in t or "next" in t:
                labels.append("on the roadmap")
            else:
                labels.append("help wanted")

    if "security" in t and "critical" not in labels:
        labels.append("critical")

    return labels


print(triage_issue("app crashes with error", []))
print(triage_issue("app crashes with error", ["bug", "needs triage"]))
print(triage_issue("add dark mode", []))
print(triage_issue("add dark mode", ["enhancement", "discussing"]))
print(triage_issue("xss security bug", []))
print(triage_issue("security vulnerability in auth", []))
print(triage_issue("easy a11y fix", ["bug", "needs triage"]))
print(triage_issue("planned api migration", ["enhancement", "discussing"]))
print(triage_issue("improve security", ["enhancement", "discussing"]))
